### ISSUE TITLE
{{expressions syntax}} does not work with the style attribute.

### ISSUE THREAD

**Issue by jpanganiban** [astoilkov/jsblocks#40]
The values doesn't seem to be interpolated in the `style` attribute. However, it works on other attributes (even with a misspelled `stye` attribute).

```
<!doctype html>
<html>
  <head>
    <script src="http://jsblocks.com/jsblocks/blocks.js"></script>
    <script>
      window.box = {};
      box.height = blocks.observable("250");
      box.width = blocks.observable("250");
      box.backgroundColor = blocks.observable("#fff");
      blocks.query(box);
    </script>
  </head>
  <body>
    Enter Height: <input type="text" data-query="val(height)" />
    Enter Width: <input type="text" data-query="val(width)" />
    Enter Background Color: <input type="text" data-query="val(backgroundColor)" />
    <h2>Height: {{height}}</h2>
    <h2>Width: {{width}}</h2>
    <h2>Background Color: {{backgroundColor}}</h2>
    <div style="background-color: {{backgroundColor}}; height: {{height}}px; width: {{width}}px;"></div>
  </body>
</html>

```

**Comment by astoilkov** [comment]
Yeah. You are correct. I never intended the `style` attribute to be evaluated because there is a `css` `data-query`. However, I agree that this brings inconsistency. For now you could use the `css` `data-query`:

```html
<div data-query="css('background-color', backgroundColor').css('height', height).css('width', width)"></div>
```

I am going to fix this.

**Comment by jpanganiban** [comment]
Cool :+1:

**Comment by Kanaye** [comment]
Okay, this one is harder to implement then I initialy thought.
Just removing the statment preventing the attributeExpressions from processing is not possible cause it would mess with the ``css data-query ``and the ``virtualElement._state``-style rendering.
Also the server side rendering seems to render this expressions, even is the client side does not.
I'll hopefully come to a solution in the next few days...

**Comment by astoilkov** [comment]
Yes. I tried to fix this a few months ago but also stumbled on these problems. I thought it would be easy to implement but it isn't. Hope you think of some solution.

**Comment by Kanaye** [comment]
Okay I have a working soloution but I have to refactor some stuff and write tests.
You can expect a pull request within the next two days.

My solution (currently) has some limitations:
The name of css-properties can (currently) not get defined with an observable:
```html
<!-- will not work -->
<div style="{{myProperty}}:{{myPropertyValue}}"></div>
```

If rendered initaly the ``css-data-query`` will override but not deleted the expression:
```javascript
blocks.query({
	expressionBackground: '#f00',
	queryBackground: '#00f'
});
```
```html
<div style="background-color: {{expressionBackground}}" data-query="css(queryBackground)"></div>
```
Will be rendered as:
```html
<div style="background-color: #00f" data-query="css(queryBackground)"></div>
```

After the initial rendering the observable that's got updated last will be the value:
```javascript
expressionBackground('#0F0'); //now the div will have the background #0f0
//... some calls (and time) later
queryBackground('#F0F'); //now it will have the background #F0F
```

**Comment by astoilkov** [comment]
This all seems logical. Can't wait to see your solution.

**Comment by Kanaye** [comment]
Quick update to inform you what's done and what's not:
- [x] Server side rendering
- [x] Clientside rendeing in Chrome, Firefox
- [x] Writing tests
- [ ] some remaning refactoring
- [ ] caching of results and not updating the DOM when nothing has changed
- [ ] Fixing IE

And the last one is the real issue. 
I developed  and tested  the [solution](https://github.com/Kanaye/jsblocks/tree/issue%2340-style-expression) with server-side rendering enabled to prevent possible issues with that. But my "good friend" the Internet Explorer has issues when *not* rendering server side.
Internet Explorer is removing invalid inline styles (like ``color: {{myColor}}``) from DOM so I can not parse them myself (as necassary in normal browser) nor I can access them with the ``style`` property of an element.
I am not sure if there is a solution for this problem.

**Comment by astoilkov** [comment]
Ok. I also investigated the IE issue and indeed I can't seem to find a way around this. The sad thing is that even the latest IE have this behavior. If we can't find a solution this would become an unimplementable feature.

**Comment by Kanaye** [comment]
I don't like to say so, but I'm not able to find a solution.
Also I'm going insane if I spent more time in IEs so called "developer tools".
I've tried many things like getting the style attribute with IE-only functions or accessing the unparsed HTML before the dom is ready and many many more.
I think that is the reason why other frameworks like angular are using their own attributes like "ng-style". 
We could easily implement a similar attribute like ``data-style`` but I don't see any benefit beside not having to write ``css()`` for every value.
But we should mention in the docs that expression are not validated in the style attribute.
